ethereum_history_api/circuits/lib/src/rlp/fragment.nr

```diff
@@ -0,0 +1,35 @@
+use crate::misc::{arrays::sub_array_equals_up_to_length, types::PaddedValue};
+use crate::rlp::decode::STRING;
+
+struct RlpFragment {
+    offset: u64,
+    length: u64,    
+    data_type: u64 // STRING or LIST
+}
+
+impl RlpFragment {
+    fn assert_string_eq<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
+        self,
+        field_name: str<FIELD_NAME_LEN>,
+        rlp: [u8; MAX_RLP_LEN],
+        value: PaddedValue<MAX_VALUE_LEN>
+    ) {
+        assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
```

Comment:
This assert checks field data type, not rlp data type, so maybe assert message should be changed